tests/integration/test_subscribe.py

```diff
@@ -251,7 +251,64 @@ def matchesRecrawledDir(subdata):
                 if not sub['is_fresh_instance']:
                     continue
                 files = self.normWatchmanFileList(sub['files'])
-                if files == ab:
+                if self.fileListsEqual(files, ab):
+                    return True
+            return False
+
+        dat = self.waitForSub('myname', root=root,
+                              accept=matchesRecrawledDir)
+        self.assertNotEqual(None, dat)
+
+        # Ensure that we observed the recrawl warning
+        warn = None
+        for item in dat:
+            if 'warning' in item:
+                warn = self.decodeBSERUTF8(item['warning'])
+                break
+        self.assertRegexpMatches(warn, r'Recrawled this watch')
+
+    # TODO: Assimilate this test into test_subscribe when Watchman gets
+    # unicode support.
+    # TODO: Correctly test subscribe with unicode on Windows.
+    @unittest.skipIf(os.name == 'nt', 'win')
```

Comment:
This doesn't exist on Python 2.6, so you'll need to import `unittest2` there instead. Check out how it's done in `WatchmanTestCase.py`.